app-ktor-server/src/main/kotlin/com/ustadmobile/lib/rest/TorrentFileRoute.kt

```diff
@@ -0,0 +1,108 @@
+package com.ustadmobile.lib.rest
+
+import com.ustadmobile.core.account.Endpoint
+import com.ustadmobile.core.contentjob.ContentJobManager
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.torrent.ContainerTorrentDownloadJob
+import com.ustadmobile.core.util.DiTag
+import com.ustadmobile.lib.db.entities.ContentJob
+import com.ustadmobile.lib.db.entities.ContentJobItem
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.kodein.di.instance
+import org.kodein.di.ktor.closestDI
+import org.kodein.di.on
+import java.io.File
+import java.lang.Exception
+
+fun Route.TorrentFileRoute(){
+
+    route("containers") {
+
+        get("{containerUid}"){
+            val torrentDir: File by closestDI().on(call).instance(tag = DiTag.TAG_TORRENT_DIR)
+            val containerUid = call.parameters["containerUid"]
+            val torrentFile = File(torrentDir, "$containerUid.torrent")
+            call.respondFile(torrentFile)
+        }
+
+        post("{containerUid}/{contentEntryUid}/upload"){
+
+            val torrentDir: File by closestDI().on(call).instance(tag = DiTag.TAG_TORRENT_DIR)
+            val contentJobManager: ContentJobManager by closestDI().instance()
+            val containerFolder: File by closestDI().on(call).instance(tag = DiTag.TAG_DEFAULT_CONTAINER_DIR)
+            val db: UmAppDatabase by closestDI().on(call).instance(tag = UmAppDatabase.TAG_DB)
+            // TODO handle virtualhostmode for endpoint
```

Comment:
This TODO is a major problem when it is running in a singleton mode. Then nothing will match. KTOR will allow you to access the config, so that you can handle this.